src/odemis/driver/xt_client.py

```diff
@@ -1275,13 +1275,24 @@ def start_autostig(self, n_images=17, sweep_range=0.015, dwell_time=3e-6, resolu
 
         Returns
         -------
-        Position
+        optimal_stigmation
             Optimal stigmator setting.
         """
         with self._proxy_access:
             self.server._pyroClaimOwnership()
-            return self.server.start_autostig(n_images, sweep_range, dwell_time, resolution,
-                                              reduced_area_size, save_results, results_fp)
+            self.server._pyroTimeout = 300  # seconds
+            try:
+                optimal_stigmation = self.server.start_autostig(n_images, sweep_range, dwell_time, resolution,
+                                                                reduced_area_size, save_results, results_fp)
+            except TimeoutError:
+                logging.warning(
+                    "Autostigmation timed out after %s s. Check the xt user interface for the current status.",
+                    self.server._pyroTimeout
+                )
+                optimal_stigmation = None
```

Comment:
I'm not convinced. Now the caller has to check whether the return is a float or None. In the case it's None, what should it do? Raise an exception? I think it'd be simpler to always return a float, or raise a TimeoutError in case of timeout. It's worth indicating that the "other side" might still be running the autostigmation as an extra warning.